web3.js/src/address-lookup-table-program.ts

```diff
@@ -0,0 +1,270 @@
+import * as BufferLayout from '@solana/buffer-layout';
+
+import * as Layout from './layout';
+import { PublicKey } from './publickey';
+import { SystemProgram } from './system-program';
+import { TransactionInstruction } from './transaction';
+import { encodeData, IInstructionInputData } from './instruction';
+
+export type CreateLookupTableParams = {
+    lookupTable: PublicKey;
+    authority: PublicKey;
+    payer: PublicKey;
+    slot: number;
+    bumpSeed: number;
+};
+
+export type FreezeLookupTableParams = {
+    lookupTable: PublicKey;
+    authority: PublicKey;
+};
+
+export type ExtendLookupTableParams = {
+    lookupTable: PublicKey;
+    authority: PublicKey;
+    payer: PublicKey;
```

Comment:
Can you make the payer optional? The reason is that if a lookup table already has enough lamports for the increased min balance, the extend instruction doesn't need a payer.